rest/config_legacy.go

```diff
@@ -236,6 +251,59 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	return &sc, dbs, nil
 }
 
+func gocbv1tov2AddressUpgrade(server string) (newServer, username, password string, err error) {
+	connSpec, err := connstr.Parse(server)
+	if err != nil {
+		return "", "", "", err
+	}
+
+	splitServers := strings.Split(server, ",")
+	// Only first host is supported to have username and pass or else gocbconnstr.Parse does not work correctly
+	if strings.Contains(splitServers[0], "@") {
+		u, err := url.Parse(splitServers[0])
+		if err != nil {
+			return "", "", "", err
+		}
+		urlUsername := u.User.Username()
+		urlPassword, urlPasswordSet := u.User.Password()
+		if urlUsername != "" && urlPasswordSet {
+			username = urlUsername
+			password = urlPassword
+		}
+	}
+
+	var hosts string
+	for _, addr := range connSpec.Addresses {
+		var port string
+		if addr.Port != 8091 && addr.Port != -1 {
+			if connSpec.Scheme != "http" {
+				port = ":" + strconv.Itoa(addr.Port)
+			} else {
+				return "", "", "", fmt.Errorf("automatic connection string conversion does not support non-default host ports. " +
+					"Please change the server field to use the couchbase(s):// scheme")
+			}
+		}
```

Comment:
Related to the `couchbase://...:8091` change suggested in the tests. I think this covers and simplifies a bit.

```suggestion
		if connSpec.Scheme == "http" && addr.Port != 8091 {
			return "", "", "", fmt.Errorf("automatic connection string conversion does not support non-default host ports. " +
				"Please change the server field to use the couchbase(s):// scheme")
		}
		if addr.Port != -1 {
			port = ":" + strconv.Itoa(addr.Port)
		}
```